src/CMakeLists.txt

```diff
@@ -10,11 +10,104 @@ include(GNUInstallDirs) # required to get a proper LIBDIR variable
 include(CMakePackageConfigHelpers)
 configure_file(base/base_uses.f90 base/base_uses.f90 @ONLY)
 
+# for the regtests
+
+set(__cp2k_ext "")
+set(__cp2k_cmake_name "")
+
+if(CP2K_USE_MPI)
+  set(__cp2k_ext "psmp")
+else()
+  set(__cp2k_ext "ssmp")
+endif()
+
+if(CP2K_USE_CUDA)
+  set(__cp2k_cmake_name "cmake_build_cuda")
+endif()
+
+if(CP2K_USE_HIP)
+  set(__cp2k_cmake_name "cmake_build_hip")
+endif()
+
+if(NOT CP2K_USE_HIP AND NOT CP2K_USE_CUDA)
+  set(__cp2k_cmake_name "cmake_build_cpu")
+endif()
+
+if(CP2K_ENABLE_REGTESTS)
+  file(MAKE_DIRECTORY "${PROJECT_SOURCE_DIR}/exe/${__cp2k_cmake_name}")
+endif()
+
+# create some short version of these libraries to ease things afterwards
+
+if(CP2K_USE_CUDA)
+  add_library(cp2k_cuda_libs INTERFACE)
+  if(CP2K_USE_NVHPC)
+    target_link_libraries(cp2k_cuda_libs INTERFACE NVHPC::MATH NVHPC::CUDA)
+  else()
+    target_link_libraries(cp2k_cuda_libs INTERFACE CUDA::cufft CUDA::cufftw
+                                                   CUDA::cublas CUDA::cudart)
+  endif()
+endif()
+
+if(CP2K_USE_HIP)
+  add_library(cp2k_hip_libs INTERFACE)
+  target_link_libraries(cp2k_hip_libs INTERFACE hip::hipfft roc::hipblas
+                                                hip::host)
+endif()
+
+set(CP2K_GPU_DFLAGS
+    $<$<BOOL:${CP2K_USE_HIP}>:__HIP_PLATFORM_AMD__
+    __OFFLOAD_HIP>
+    $<$<COMPILE_LANGUAGE:HIP>:__HIP_PLATFORM_AMD__
+    __OFFLOAD_HIP>
+    $<$<BOOL:${CP2K_USE_CUDA}>:__OFFLOAD_CUDA>
+    $<$<COMPILE_LANGUAGE:CUDA>:__OFFLOAD_CUDA>)
+
+set(CP2K_DBM_DFLAGS
+    $<$<BOOL:${CP2K_DISABLE_DBM_GPU}>:__NO_OFFLOAD_DBM>
+    $<$<NOT:$<BOOL:${CP2K_DISABLE_DBM_GPU}>>:${CP2K_GPU_DFLAGS}>
+    $<$<BOOL:${CP2K_USE_LIBXSMM}>:__LIBXSMM>)
+
+set(CP2K_GRID_DFLAGS
+    $<$<BOOL:${CP2K_DISABLE_GRID_GPU}>:__NO_OFFLOAD_GRID>
+    $<$<NOT:$<BOOL:${CP2K_DISABLE_GRID_GPU}>>:${CP2K_GPU_DFLAGS}>
+    $<$<BOOL:${CP2K_USE_LIBXSMM}>:__LIBXSMM>
+    $<$<STREQUAL:${CP2K_BLAS_VENDOR},"MKL">:__MKL>)
```

Comment:
Do we actually need this special treatment? Can't we just include `__NO_OFFLOAD_DBM` and `__NO_OFFLOAD_GRID` in the target_compile_definitions for all targets?